subsquid/src/processors/pablo.ts

```diff
@@ -319,37 +323,38 @@ export async function processLiquidityRemovedEvent(ctx: Context, block: Block, e
   await ctx.store.save(pool);
 
   // Update or create assets
-  for (const [assetId, amount] of assetAmounts) {
-    const asset = await getOrCreatePabloAsset(ctx, block, pool, assetId.toString());
-
-    asset.totalLiquidity -= amount;
-    asset.blockId = block.header.hash;
-
-    await ctx.store.save(asset);
-
-    const historicalLockedValue = new HistoricalLockedValue({
-      id: randomUUID(),
-      event,
-      amount: -amount,
-      accumulatedAmount: asset.totalLiquidity,
-      timestamp: new Date(block.header.timestamp),
-      source: LockedSource.Pablo,
-      assetId: assetId.toString(),
-      sourceEntityId: pool.id,
-      blockId: block.header.hash
-    });
-
-    await ctx.store.save(historicalLockedValue);
-
-    await getOrCreateHistoricalAssetPrice(ctx, assetId.toString(), block.header.timestamp);
-  }
-
   const amounts: Array<PabloAmount> = [];
 
   for (const [assetId, amount] of assetAmounts) {
-    const price = await getOrCreateHistoricalAssetPrice(ctx, assetId.toString(), block.header.timestamp);
-    const pabloAmount = new PabloAmount({ assetId: assetId.toString(), amount, price: price || 0 });
-    amounts.push(pabloAmount);
+    try {
+      const asset = await getOrCreatePabloAsset(ctx, block, pool, assetId.toString());
+
+      asset.totalLiquidity -= amount;
+      asset.blockId = block.header.hash;
+
+      await ctx.store.save(asset);
+
+      const historicalLockedValue = new HistoricalLockedValue({
+        id: randomUUID(),
+        event,
+        amount: -amount,
```

Comment:
How about considering it to write `amount * -1` ? (It has the same result, just improve the readability.

Reply:
Done 👌🏼